### ISSUE TITLE
Wacom pen not working in fullscreen

### ISSUE THREAD

**Issue by lokman2k5** [flxzt/rnote#1172]
**Describe the bug**  
When I go into fullscreen mode, my laptop pen stops working. Other types of input e.g. fingers, still work but the pen stops completely. It works fine in windowed mode. 

**To Reproduce**  
1. Go into fullscreen mode (F11)
2. try writing something or clicking a button with the pen 

**Expected behavior**  
the pen should work normally, like in windowed mode

**Console Output**  
2024-08-10T23:45:26.710112Z DEBUG rnote: .. tracing subscriber initialized.

**Desktop (please complete the following information):**  
 - OS: Ubuntu 24.04LTS
 - App Version: 0.11.0
 - Installation Source: Flathub
 - Desktop Environment: Gnome 46
 - Display Server: Wayland
 - Input Source: On-screen wacom tablet (a 2-in-1 laptop with a touchscreen and a pen)

**Additional context**  
Usually, I can still use it in fullscreen mode until I press any button like a color, it then stops and I have to get out of fullscreen mode then in again. Even if I enable "Draw with touch input" and use fingers instead of pen, it works just fine, the only issue it's having is with the laptop pen.

**Comment by Doublonmousse** [comment]
Does this happen with any other gtk app (that you can fullscreen) like nautilus ?
This _feels_ like a gnome/mutter bug. 
There is one issue report on that exact issue here https://gitlab.gnome.org/GNOME/mutter/-/issues/3616

Given the timing (of your report and the one on the mutter repo), chances are it's a recent regression of mutter. Maybe downgrade mutter in the meantime (this may also be useful for debugging purposes, if known working and non working issues are known).

**Comment by mahdihasan2006** [comment]
I have the same issue. But it happens for all apps in Fedora 40. However, recently I tried Ubuntu and the I didn't see the problem there.

**Comment by Doublonmousse** [comment]
@mahdihasan2006 Fedora 40 with gnome right ?
Can you get the mutter or gnome DE version for both systems (and whether they are running with wayland or X11 ?).

The ubuntu version is probably older than the fedora version, given 24.04 is also an LTS

**Comment by mahdihasan2006** [comment]
> Fedora 40 with gnome right ?

Yes.

Mutter version `46.4` in Fedora. In Ubuntu I got this output from terminal:
```sh
mahdi@ubuntu:~$ mutter --version
Command 'mutter' not found, but can be installed with:
sudo apt install mutter
```
But I updated Ubuntu today and the tablet stopped working in fullscreen mode of any app like Fedora. - Using **Wayland**

Both in Fedora and Ubuntu, tablet works in fullscreen mode while using **X11**. But pen pressure doesn't work in Rnote in **X11**.

![image](https://github.com/user-attachments/assets/7959e08f-8644-461d-b607-d99d1f5ea27f)

`gir1.2-mutter-14`, `libmutter-14-0`, `mutter-common`, `mutter-common-bin` was updated from `46.0` to `46.2`.

**Extra Information**
I am using [XP-PEN Deco Fun L](https://www.xp-pen.com/product/deco-fun-xs-s-l.html) and [OpenTabletDriver](https://opentabletdriver.net/).

**Comment by AesaraB** [comment]
@lokman2k5 @mahdihasan2006 How many displays do you have? Is your mapped display the leftmost one?

**Comment by mahdihasan2006** [comment]
> How many displays do you have? Is your pen tablet mapped to the leftmost display?

I am using a laptop with only one display.

**Comment by AesaraB** [comment]
> > How many displays do you have? Is your pen tablet mapped to the leftmost display?
> 
> I am using a laptop with only one display.

Try connect another display and see if the problem happens when you try to use the tablet on both screens, or only one.

**Comment by mahdihasan2006** [comment]
I don't have any external display.

I am using an XP-PEN tablet and the Wacom Tablet settings is empty in Gnome Settings app.

![image](https://github.com/user-attachments/assets/203fb041-03b1-414a-8d12-1b3ab5fceabf)

By the way, I haven't tried the official XP-PEN drivers since the problem started.

**Comment by AesaraB** [comment]
You're probably running libwacom <2.12.0, when it gets updated the settings page should be okay.. Unfortunately without another display it's impossible to test the bug's conditions.

**Comment by StoneToken** [comment]
Hi! Got same problem on Fedora 40 + wacom based stylus (lenovo x1 yoga gen1). If Rnote is fullscreen stylus stops working after 1-2 strokes. If I exit full screen or switch to other app the stylus get back to work until I continue in full screen.
Environment
libwacom 2.11.0-1.fc40
single screen

If needed I can do some testing \ loggng.

**Comment by lokman2k5** [comment]
> @lokman2k5 @mahdihasan2006 How many displays do you have? Is your pen tablet mapped to the leftmost display?

I have one display. Thinkpad x1 yoga g1

**Comment by lokman2k5** [comment]
> Does this happen with any other gtk app (that you can fullscreen) like nautilus ? This _feels_ like a gnome/mutter bug. There is one issue report on that exact issue here https://gitlab.gnome.org/GNOME/mutter/-/issues/3616
> 
> Given the timing (of your report and the one on the mutter repo), chances are it's a recent regression of mutter. Maybe downgrade mutter in the meantime (this may also be useful for debugging purposes, if known working and non working versions are known).

I tried with another application (xournalpp) and can confirm the issue is there.. Is there any way around this in the mean time?

**Comment by Doublonmousse** [comment]
The upstream issue here : https://gitlab.gnome.org/GNOME/mutter/-/issues/3616 has been closed recently.

So it's probably more of a matter of time before this is rolled out in the next mutter update

**Comment by lokman2k5** [comment]
This seems to be fixed in mutter 46.5, but I am starting to see something else: 
![Screenshot from 2024-09-18 10-47-59](https://github.com/user-attachments/assets/829afcb9-bb89-47f2-8728-3c2e24486c77)
this can happen in rnote and many other applications, if you are experiencing such issue please say so on the bug report I made over here: https://gitlab.gnome.org/GNOME/mutter/-/issues/3679
